R/metadata.R

```diff
@@ -0,0 +1,70 @@
+#' Create cache_meta table
+
+CACHE_METADATA_TABLE <- "cache_metadata"
+
+#' Track cache metadata and register salt if needed.
+#'
+#' @param dbconn SQLConnection. A database connection.
+#' @param table_name character.  Name of cache table.
+#' @param salt list. Actual value of salt before hashing.
+track_cache_salt <- function(dbconn, table_name, salt) {
+
+  if (!DBI::dbExistsTable(dbconn, CACHE_METADATA_TABLE)) {
+    DBI::dbGetQuery(dbconn, paste0("CREATE TABLE ", CACHE_METADATA_TABLE,
+      " (table_name varchar(255) UNIQUE NOT NULL, salt_obj text NOT NULL)"))
+  }
+
+  df <- DBI::dbGetQuery(dbconn, paste0("SELECT * from ", CACHE_METADATA_TABLE, " WHERE table_name = '", table_name, "'"))
+
+  salt_obj <- serialize_to_string(salt)
+
+  if (NROW(df) > 0) {
```

Comment:
Good point we should never have more than one row matching the table names.  It's also already handled by setting a UNIQUE index on `talble_name` in L14.